src/I18nBundle/Adapter/Redirector/GeoRedirector.php

```diff
@@ -53,40 +53,49 @@ public function makeDecision(RedirectorBag $redirectorBag)
             return;
         }
 
-        $userLanguageIso = $this->userHelper->guessLanguage();
+        $userLanguagesIso = $this->userHelper->getLanguagesAcceptedByUser();
         $userCountryIso = false;
 
-        $url = null;
-
-        $redirectorOptions = [
-            'geoLanguage' => $userLanguageIso,
-            'geoCountry'  => $userCountryIso,
-        ];
-
-        if ($userLanguageIso === false) {
-            $this->setDecision(['valid' => false, 'redirectorOptions' => $redirectorOptions]);
+        if (count($userLanguagesIso) === 0) {
+            $this->setDecision(['valid' => false, 'redirectorOptions' => [
+                'geoLanguage' => false,
+                'geoCountry'  => $userCountryIso,
+            ]]);
 
             return;
         }
 
-        if ($redirectorBag->getI18nMode() === 'country') {
-            $userCountryIso = $this->userHelper->guessCountry();
-            $redirectorOptions['geoCountry'] = $userCountryIso;
-            if ($userCountryIso === false) {
-                $this->setDecision(['valid' => false, 'redirectorOptions' => $redirectorOptions]);
 
-                return;
+        foreach($userLanguagesIso as $userLanguageIso) {
+            $url = null;
+
+            $redirectorOptions = [
+                'geoLanguage' => $userLanguageIso,
+                'geoCountry'  => $userCountryIso,
+            ];
+
+            if ($redirectorBag->getI18nMode() === 'country') {
+                $userCountryIso = $this->userHelper->guessCountry();
+                $redirectorOptions['geoCountry'] = $userCountryIso;
+                if ($userCountryIso === false) {
+                    $this->setDecision(['valid' => false, 'redirectorOptions' => $redirectorOptions]);
+
+                    return;
+                }
             }
-        }
 
-        if ($redirectorBag->getI18nMode() === 'language') {
-            $url = $this->findUrlInZoneTree($userLanguageIso);
-        } elseif ($redirectorBag->getI18nMode() === 'country') {
-            $userCountryIso = $this->userHelper->guessCountry();
-            $url = $this->findUrlInZoneTree($userLanguageIso, $userCountryIso);
-        }
+            if ($redirectorBag->getI18nMode() === 'language') {
+                $url = $this->findUrlInZoneTree($userLanguageIso);
+            } elseif ($redirectorBag->getI18nMode() === 'country') {
+                $userCountryIso = $this->userHelper->guessCountry();
+                $url = $this->findUrlInZoneTree($userLanguageIso, $userCountryIso);
+            }
 
-        $valid = !empty($url);
+            $valid = !empty($url);
+            if($valid) {
```

Comment:
@BlackbitNeueMedien use `if(!empty($url))` instead of creating a new variable `$valid` (i know, it's my sloppy code)